### PR TITLE
feat!: remove appbartab and make headertab simpler/accessible

### PR DESCRIPTION
**Pull request by borisonr**
## Title

Back with more headertab experimentation...
re: https://codecademy.atlassian.net/browse/A11Y-369

this pr removes the appbartab too which was only used in the footer

the headertab is now really just a div with margin right... so probably not super useful. open to ideas for what to do from here

monolith implementation of this: https://github.com/codecademy-engineering/Codecademy/pull/17448

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.5.11-alpha.d707d3.0<br/>@codecademy/gamut@7.6.1-alpha.d707d3.0<br/></pre>

File: packages/brand-components/src/Header/HeaderTab/index.tsx [hunk 1]
```diff
@@ -3,40 +3,21 @@ import cx from 'classnames';
...
-      onClick={handleClick}
-      className={classes}
-      onKeyDown={(e: React.KeyboardEvent) => {
-        // https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_link_role
-        if (e.key === ' ' || e.key === 'Enter') {
-          handleClick(e);
-        }
-      }}
-    >
+    <div id={id} className={classes} data-testid={testId}>
```
Comment (JoshuaKGoldberg) [inline packages/brand-components/src/Header/HeaderTab/index.tsx:20]: I forget, was there a reason for `id={id}`?

Comment (borisonr) [reply packages/brand-components/src/Header/HeaderTab/index.tsx:20]: most of the headertabs in a monolith pass in an id... unclear why?

Comment (borisonr) [reply packages/brand-components/src/Header/HeaderTab/index.tsx:20]: maybe for responsive design

Comment (borisonr) [reply packages/brand-components/src/Header/HeaderTab/index.tsx:20]: or maybe it's legacy and can be deleted...

Comment (JoshuaKGoldberg) [reply packages/brand-components/src/Header/HeaderTab/index.tsx:20]: 🤷 my 💰 is on legacy, from before we consolidated on `data-testid`

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.5.12-alpha.e4f9e9.0<br/>@codecademy/gamut@7.7.1-alpha.e4f9e9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eb9572a89c051b03034797d--codecademy-styleguide.netlify.app>https://5eb9572a89c051b03034797d--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eb9572a89c051b03034797d>Deploy Logs</a></p>

File: packages/brand-components/src/Header/HeaderTab/index.tsx (hunk 1 above)
Comment (borisonr) [reply packages/brand-components/src/Header/HeaderTab/index.tsx:20]: forgot to follow up on this, but i'm going to leave the id in here for now and remove the ids from the consumers first to make sure styles dont randomly break or anything

**Pull request merged**